Code/src/main/java/nl/utwente/group10/ui/BackendUtils.java

```diff
@@ -31,4 +33,18 @@ public static Type dive(ConstT function, int steps) {
         }
         return function;
     }
+    
+    /**
+     * @return Whether or not t1 and t2 unify with each other.
```

Comment:
```unify()``` should indeed never be called from the front end.

The front end should not test whether types match. The front end should just apply things and call analyze appropriately. Then the type checker will tell you if there is something wrong or not.